java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt

```diff
@@ -5442,11 +5445,15 @@ open class KotlinFileExtractor(
     /**
      * Extracts the class around a local function or a lambda. The superclass must have a no-arg constructor.
      */
-    private fun extractGeneratedClass(localFunction: IrFunction, superTypes: List<IrType>) : Label<out DbClass> {
+    private fun extractGeneratedClass(
+        localFunction: IrFunction,
+        superTypes: List<IrType>,
+        compilerGeneratedKind: CompilerGeneratedKinds? = null
```

Comment:
Is there a case when `extractGeneratedClass` is called and the class should be non compiler generated? Should this parameter be nullable?

Reply:
I assume that any class extracted by "extractGeneratedClass" should be treated as compiler-generated. As in the other comment, `null` here really means "the default".